libqtile/state.py

```diff
@@ -45,6 +53,39 @@ def apply(self, qtile):
             Rearrange the windows in the specified Qtile object according to
             this QtileState.
         """
+        try:
+            for group in qtile.groups:
+                x = []
+                for i in self.focusHistory[group.name]:
+                    x.append(qtile.windowMap[i.Wid])
+                group.focusHistory = x
+                for layout in group.layouts:
```

Comment:
It looks like all this is some code to pickle by hand. Is there any reason for not just using pickle directly? This is where your `__getstate__` and `__setstate__` methods would be used; I think they're unused now.